src/WebJobs.Script/Workers/Rpc/WebHostRpcWorkerChannelManager.cs

```diff
@@ -294,25 +295,34 @@ public Task ShutdownChannelsAsync()
                 {
                     foreach (string workerId in standbyChannels.Keys)
                     {
-                        standbyChannels[workerId]?.Task.ContinueWith(channelTask =>
```

Comment:
This was all previously fire-and-forget with the way the continuation was wired up. While it wasn't a direct cause of what we were seeing, I realized that there was no guaranteed way to wait for channels to be shut down without refactoring this a bit.

I checked all callers and some were already debouncing or fire-and-forgetting the call -- so this should not affect behavior.